flaky test TestUnionIssue

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
 --- FAIL: TestUnionIssue (5.20s)
    testkit.go:123: RefreshSession rand seed: 1733290409967143946
    result.go:49: 
        	Error Trace:	pkg/testkit/result.go:49
        	            				pkg/executor/test/issuetest/executor_issue_test.go:98
        	Error:      	Not equal: 
        	            	expected: "[31]\n[1]\n"
        	            	actual  : "[1]\n[31]\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	+[1]
        	            	 [31]
        	            	-[1]
        	            	 
        	Test:       	TestUnionIssue
        	Messages:   	sql:select hex(cast('1' as char)) union all select '1';, args:[]
FAIL
```
### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/executor/test/issuetest/executor_issue_test.go
+++ b/pkg/executor/test/issuetest/executor_issue_test.go
@@ -93,9 +93,9 @@ func TestUnionIssue(t *testing.T) {
 	tk.MustExec("create table t1 (id int);")
 	tk.MustExec("create table t2 (id int, c int);")
 	// Issue56587
-	tk.MustQuery("select quote(cast('abc' as char)) union all select '1'").Check(testkit.Rows("'abc'", "1"))
+	tk.MustQuery("(select quote(cast('abc' as char)) union all select '1') order by 1").Check(testkit.Rows("'abc'", "1"))
 	tk.MustQuery(`select elt(2, "1", cast('abc' as char)) union all select "12" where false`).Check(testkit.Rows("abc"))
-	tk.MustQuery(`select hex(cast('1' as char)) union all select '1';`).Check(testkit.Rows("31", "1"))
+	tk.MustQuery(`(select hex(cast('1' as char)) union all select '1') order by 1;`).Check(testkit.Rows("1", "31"))
 
 	testCases := []struct {
 		sql    string

--- a/pkg/executor/test/issuetest/executor_issue_test.go
+++ b/pkg/executor/test/issuetest/executor_issue_test.go
@@ -93,9 +93,9 @@ func TestUnionIssue(t *testing.T) {
 	tk.MustExec("create table t1 (id int);")
 	tk.MustExec("create table t2 (id int, c int);")
 	// Issue56587
-	tk.MustQuery("(select quote(cast('abc' as char)) union all select '1') order by 1").Check(testkit.Rows("'abc'", "1"))
+	tk.MustQuery("select quote(cast('abc' as char)) union all select '1'").Sort().Check(testkit.Rows("'abc'", "1"))
 	tk.MustQuery(`select elt(2, "1", cast('abc' as char)) union all select "12" where false`).Check(testkit.Rows("abc"))
-	tk.MustQuery(`(select hex(cast('1' as char)) union all select '1') order by 1;`).Check(testkit.Rows("1", "31"))
+	tk.MustQuery(`select hex(cast('1' as char)) union all select '1'`).Sort().Check(testkit.Rows("1", "31"))
 
 	testCases := []struct {
 		sql    string

